integrations/tensorflow/e2e/concat_test.py

```diff
@@ -0,0 +1,91 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test concat op."""
+
+from pyiree.tf.support import tf_test_utils
+import tensorflow.compat.v2 as tf
+
+
+class ConcatOpsModule(tf.Module):
+
+  @tf.function(input_signature=[
+      tf.TensorSpec([1, 5, 0], tf.float32),
+      tf.TensorSpec([1, 5, 1], tf.float32),
+  ])
+  def concat_zero_dim(self, a, b):
+    return tf.concat([a, b], axis=2)
+
+  @tf.function(input_signature=[
+      tf.TensorSpec([1, 5, 1], tf.float32),
+      tf.TensorSpec([1, 5, 1], tf.float32),
+  ])
+  def concat0axis(self, a, b):
+    return tf.concat([a, b], axis=0)
+
+  @tf.function(input_signature=[
+      tf.TensorSpec([1, 5, 1], tf.float32),
+      tf.TensorSpec([1, 5, 1], tf.float32),
+  ])
+  def concat1axis(self, a, b):
+    return tf.concat([a, b], axis=1)
+
+  @tf.function(input_signature=[
+      tf.TensorSpec([1, 5, 1], tf.float32),
+      tf.TensorSpec([1, 5, 1], tf.float32),
+  ])
+  def concat2axis(self, a, b):
+    return tf.concat([a, b], axis=2)
+
+
+@tf_test_utils.compile_modules(mat=ConcatOpsModule)
+class ConcatOpsTest(tf_test_utils.SavedModelTestCase):
+
+  def test_concat_zero_dim(self):
+    tf_test_utils.set_random_seed()
+    m = self.modules.mat.all
+    a = tf.random.uniform([1, 5, 0], dtype=tf.float32)
```

Comment:
This is actually will result an empty tensors.

Reply:
Tensor shape will be: TensorShape([Dimension(1), Dimension(5), Dimension(0)])